src/join/join.rs

```diff
@@ -0,0 +1,374 @@
+#![crate_name = "uu_join"]
+
+/*
+ * This file is part of the uutils coreutils package.
+ *
+ * (c) Konstantin Pospelov <kupospelov@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+extern crate clap;
+
+#[macro_use]
+extern crate uucore;
+
+use std::fs::File;
+use std::io::{BufRead, BufReader, Lines, Stdin, stdin};
+use std::cmp::Ordering;
+use clap::{App, Arg};
+
+static NAME: &'static str = "join";
+static VERSION: &'static str = env!("CARGO_PKG_VERSION");
+
+#[derive(PartialEq)]
+enum FileNum {
+    None,
+    File1,
+    File2,
+}
+
+struct Settings {
+    key1: usize,
+    key2: usize,
+    print_unpaired: FileNum,
+    ignore_case: bool,
+}
+
+impl Default for Settings {
+    fn default() -> Settings {
+        Settings {
+            key1: 0,
+            key2: 0,
+            print_unpaired: FileNum::None,
+            ignore_case: false,
+        }
+    }
+}
+
+struct Line {
+    fields: Vec<String>,
+}
+
+impl Line {
+    fn new(string: String) -> Line {
+        Line { fields: string.split_whitespace().map(|s| String::from(s)).collect() }
```

Comment:
Oh, I think I forgot to mention this, but `.map(|s| String::from(s))` could just be `.map(String::from)`.